pkg/test/client.go

```diff
@@ -42,37 +50,51 @@ func (c *FakeClient) Get(ctx context.Context, key client.ObjectKey, obj runtime.
 	return c.Client.Get(ctx, key, obj)
 }
 
-func (c *FakeClient) List(ctx context.Context, opts *client.ListOptions, list runtime.Object) error {
+func (c *FakeClient) List(ctx context.Context, list runtime.Object, opts ...client.ListOption) error {
 	if c.MockList != nil {
-		return c.MockList(ctx, opts, list)
+		return c.MockList(ctx, list, opts...)
 	}
-	return c.Client.List(ctx, opts, list)
+	return c.Client.List(ctx, list, opts...)
 }
 
-func (c *FakeClient) Create(ctx context.Context, obj runtime.Object) error {
+func (c *FakeClient) Create(ctx context.Context, obj runtime.Object, opts ...client.CreateOption) error {
 	if c.MockCreate != nil {
-		return c.MockCreate(ctx, obj)
+		return c.MockCreate(ctx, obj, opts...)
 	}
-	return c.Client.Create(ctx, obj)
+	return c.Client.Create(ctx, obj, opts...)
 }
 
 func (c *FakeClient) Status() client.StatusWriter {
+	m := mockStatusUpdate{}
+	if c.MockStatusUpdate == nil && c.MockStatusPatch == nil {
+		return c.Client.Status()
+	}
+	if c.MockStatusUpdate != nil {
+		m.mockUpdate = c.MockStatusUpdate
+	}
 	if c.MockStatusUpdate != nil {
```

Comment:
shouldn't it be
```suggestion
	if c. MockStatusPatch != nil {
		m. mockPatch = c. MockStatusPatch
	}
```

Reply:
(be careful though, I'm not sure how this suggestion will affect the current code, in terms of lines to change)

Reply:
Yes. Good catch. Fixed.